src/main/java/org/assertj/core/api/AbstractClassAssert.java

```diff
@@ -317,7 +317,12 @@ public SELF isPackagePrivate() {
    * @throws AssertionError if {@code actual} is {@code null}.
    * @throws AssertionError if the actual {@code Class} doesn't contains all of these annotations.
    */
-  public SELF hasAnnotations(@SuppressWarnings("unchecked") Class<? extends Annotation>... annotations) {
+  @SafeVarargs
+  public final SELF hasAnnotations(@SuppressWarnings("unchecked") Class<? extends Annotation>... annotations) {
+    return hasAnnotationsForProxy(annotations);
+  }
+
+  protected SELF hasAnnotationsForProxy(Class<? extends Annotation>[] annotations) {
```

Comment:
would be good to add a comment explaining why we have the `xxxForProxy` methods, if you feel duplicating the comment n times is a bit too much I'm happy to have it at the class level.

Reply:
Makes sense. I've added a comment to all methods. Please have a look at it and let me know if it is enough